src/imitation/util/logger.py

```diff
@@ -0,0 +1,126 @@
+import contextlib
+import os
+from typing import Dict, Optional, Sequence
+
+from stable_baselines.common.misc_util import mpi_rank_or_zero
+import stable_baselines.logger as sb_logger
+
+
+def _sb_logger_configure_replacement(*args, **kwargs):
+  raise RuntimeError("Shouldn't call stable_baselines.logger.configure "
+                     "once imitation.logger.configure() has been called")
+
+
+def _sb_logger_reset_replacement():
+  raise NotImplementedError("reset() while using imitation.configure()")
+
+
+_in_accumul_context = False
+_configured = False
+_format_strs = None
+
+
+def configure(folder: str, format_strs: Optional[Sequence[str]] = None) -> None:
+  """Configure Stable Baselines logger to be `accumulate_means()`-compatible.
+
+  After this function is called, `stable_baselines.logger.{configure,reset}()`
+  are replaced with stubs that raise errors (not yet implemented).
+
+  Args:
+      folder: Argument from `stable_baselines.logger.configure`.
+      format_strs: Argument from `stable_baselines.logger.configure`.
+  """
+  global _configured, _format_strs
+  assert not _in_accumul_context
+  _configured = True
+  _format_strs = format_strs
+
+  sb_logger.configure = _sb_logger_configure_replacement
+  sb_logger.reset = _sb_logger_reset_replacement
+
+  output_formats = _build_output_formats(folder, format_strs)
+  sb_logger.Logger.DEFAULT = sb_logger.Logger(folder, output_formats)
+  sb_logger.Logger.CURRENT = sb_logger.Logger.DEFAULT
+  sb_logger.log('Logging to %s' % folder)
+
+
+def dumpkvs():
+  """Alias for `stable_baselines.logger.logkv`."""
+  sb_logger.dumpkvs()
+
+
+@contextlib.contextmanager
+def accumulate_means(subdir_name: str):
+  """Temporarily redirect logkv() to a different logger and auto-track kvmeans.
+
+  Within this context, the original logger is swapped out for a special logger
+  in directory `"{current_logging_dir}/raw/{subdir_name}"`.
+
+  The special logger's `stable_baselines.logger.logkv(key, val)`, in addition
+  to tracking its own logs, also forwards the log to the original logger's
+  `.logkv_mean()` under the key `mean/{subdir_name}/{key}`.
+
+  After the context exits, these means can be dumped as usual using
+  `stable_baselines.logger.dumpkvs()` or `imitation.util.logger.dumpkvs()`.
+
+  This context cannot be nested.
+
+  Args:
+    subdir_name: Chooses the logger subdirectories and temporary logger.
+  """
+  global _in_accumul_context
+  assert _configured
+  assert not _in_accumul_context
+  _in_accumul_context = True
+
+  try:
+    sb_logger.Logger.CURRENT = _AccumulatingLogger.from_subdir(subdir_name)
+    yield
```

Comment:
Don't think that we should yield the `_AccumulatingLogger`. In my eyes this is a private class that the user shouldn't be messing with. Instead they should be interacting with `sb_logger` as usual.

I suppose that a helpful thing about yielding `_AccumulatingLogger` is that you wouldn't need to import `sb_logger` in addition to `imitation.util.logger` if you wanted to `logkv` yourself inside the `accumulate_means` context that you just made (right now we don't `sb_logger.logkv` in AdversarialTrainer because PPO2 does all of this).

We could avoid needing both imports in this case by simply importing all the `logkv`, ..., `debug`, `info`, `warning` utility methods from `sb_logger` straight into `imitation.util.logger`.

Was this the `yield` use case you were thinking about?